app/validators/open_before_closed_validator.rb

```diff
@@ -0,0 +1,15 @@
+class OpenBeforeClosedValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, _value)
+    record.errors.add(attribute, error_message) unless before_closed?(record)
+  end
+
+private
+
+  def before_closed?(record)
+    record.opened_date <= record.closed_date || record.closed_date.blank?
+  end
```

Comment:
that seems a bit dangerous on documents that have already been published/created with invalid dates. Do any documents with invalid data currently exist? I think having the check for nil to avoid an unnecessary 500 is probably worthwhile.